pkg/promtail/targets/filetarget.go

```diff
@@ -198,12 +159,104 @@ func (t *FileTarget) run() {
 			}
 		case err := <-t.watcher.Errors:
 			level.Error(t.logger).Log("msg", "error from fswatch", "error", err)
+		case <-ticker.C:
+			err := t.sync()
+			if err != nil {
+				level.Error(t.logger).Log("msg", "error running sync function", "error", err)
+			}
 		case <-t.quit:
 			return
 		}
 	}
 }
 
+func (t *FileTarget) sync() error {
+
+	// Find list of directories to add to watcher.
+	var err error
```

Comment:
This `var err error` is redundant, the line below will instantiate the variable too.